### ISSUE TITLE
Deriving `Deserialize` with container `content` attribute causes `unused_results` lint

### ISSUE THREAD

**Issue by zheland** [serde-rs/serde#2119]
Minimal example:
```rust
#![deny(unused_results)]

#[derive(serde::Deserialize)]
#[serde(tag = "mode", content = "value")]
pub enum Connection {
    Foo(String),
    Bar(String),
}
```
Output:
```
   Compiling playground v0.0.1 (/playground)
error: unused result
 --> src/lib.rs:3:10
  |
3 | #[derive(serde::Deserialize)]
  |          ^^^^^^^^^^^^^^^^^^
  |
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![deny(unused_results)]
  |         ^^^^^^^^^^^^^^
  = note: this error originates in the macro `try` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unused result
 --> src/lib.rs:3:10
  |
3 | #[derive(serde::Deserialize)]
  |          ^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `try` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `playground` due to 5 previous errors
```

Error does not occur for Serialize and does not occur when using only the `tag` attribute without the `content` container attribute.

**Comment by d-e-s-o** [comment]
This issue seems to have been fixed with `serde` `1.0.131`. I can reproduce with 
```toml
[dependencies]
serde = {version = "=1.0.130", features = ["derive"]}
```
but no longer with
```toml
[dependencies]
serde = {version = "=1.0.131", features = ["derive"]}
```
Can you confirm @zheland ?

**Comment by zheland** [comment]
Yes, that's fixed in `1.0.131`, thank you!
